acir/src/circuit/opcodes/black_box_function_call.rs

```diff
@@ -71,6 +71,11 @@ pub enum BlackBoxFuncCall {
         inputs: Vec<FunctionInput>,
         outputs: Vec<Witness>,
     },
+    Keccak256VariableLength {
+        inputs: Vec<FunctionInput>,
+        var_message_size: FunctionInput,
```

Comment:
Could we specify what `var_message_size` represents? Is it the number of inputs to accept? Or is it the number of bytes of the full message (I know from pwg/hash.rs that it is the num bytes but this is unclear to me on its own)?

Reply:
Good point -- I'll add a comment

Reply:
Something else to add is a safety check for when `var_message_size` is more than the input bytes -- We could define the behaviour as taking all of the message of just returning an error. I'll return an error as its the safer option and users will need to constrain `var_message_size` appropriately.